### PR TITLE
Fix: failing assert in bin/console state-machine:dump

### PR DESCRIPTION
**Pull request by amenk**
```
bin/console state-machine:dump order.state
```

fails

assert(\is_string($v))

Because there is a 'label' => null entry in the options array.

### 5. Checklist

- [x] I have rebased my changes to remove merge conflicts
- [ ] I have written tests and verified that they fail without my change
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [ ] I have read the contribution requirements and fulfil them.

### CODE REVIEW

**Comment by amenk** [comment]
Shall I create a changelog entry for this quite minor change?

File: src/Core/System/StateMachine/Util/StateMachineGraphvizDumper.php
Comment (AydinHassan) [inline src/Core/System/StateMachine/Util/StateMachineGraphvizDumper.php]: I don't think this is correct for the key? can keys even be null?

Comment (shyim) [reply src/Core/System/StateMachine/Util/StateMachineGraphvizDumper.php]: php arrays keys can be null

**Comment by AydinHassan** [comment]
@amenk no need for changelog imo. But I'm not sure this is the correct fix, I think I would rather leave out the label option when it's null at the call site. But I will defer to the team who is responsible for this area :)

File: src/Core/System/StateMachine/Util/StateMachineGraphvizDumper.php
Comment (AydinHassan) [reply src/Core/System/StateMachine/Util/StateMachineGraphvizDumper.php]: weird, so they can. Not correct here tho imo.

Comment (mitelg) [reply src/Core/System/StateMachine/Util/StateMachineGraphvizDumper.php]: they are not really `null` then, are they? :thinking: aren't they implicitly cast to an empty string?

Comment (AydinHassan) [reply src/Core/System/StateMachine/Util/StateMachineGraphvizDumper.php]: yeah: https://3v4l.org/sMZjS#v8.3.11

Comment (amenk) [reply src/Core/System/StateMachine/Util/StateMachineGraphvizDumper.php]: good point

**Comment by amenk** [comment]
@AydinHassan okay, thanks for the feedback. Feel free to take over in the responsible team :-)

**Comment by cyl3x** [comment]
sprint's `%s` will replace `null` values with an empty string anyway, so properly omitting `null` values is preferable. Also, this way, you can remove options and still use an explicit empty string if you wish

**Comment by amenk** [comment]
How do we proceed with this? It's minor thing though :-)

**Comment by cyl3x** [comment]
@amenk Hey, I'm still waiting for the review changes :smile: :
- Adding a changelog
- Omitting `null` values like Aydin suggested

**Comment by amenk** [comment]
> Hey, I'm still waiting for the review changes 😄 :
> 
>     * Adding a changelog
> 
>     * Omitting `null` values like Aydin suggested

done

File: changelog/_unreleased/2024-12-06-fix-assert-in-state-machine-dump.md
Comment (cyl3x) [inline changelog/_unreleased/2024-12-06-fix-assert-in-state-machine-dump.md]:
```suggestion
```

File: changelog/_unreleased/2024-12-06-fix-assert-in-state-machine-dump.md
Comment (cyl3x) [inline changelog/_unreleased/2024-12-06-fix-assert-in-state-machine-dump.md]:
```suggestion
```

File: changelog/_unreleased/2024-12-06-fix-assert-in-state-machine-dump.md
Comment (cyl3x) [inline changelog/_unreleased/2024-12-06-fix-assert-in-state-machine-dump.md]:
```suggestion
```

File: src/Core/System/StateMachine/Util/StateMachineGraphvizDumper.php
Comment (cyl3x) [inline src/Core/System/StateMachine/Util/StateMachineGraphvizDumper.php]: So that `null` values are omitted:
```suggestion
            if ($v !== null) {
                \assert(\is_scalar($v) || $v instanceof \Stringable);
                $code[] = \sprintf('%s="%s"', $k, $v);
            }
```

**Review by cyl3x — CHANGES_REQUESTED** [review]

**Comment by cyl3x** [comment]
@amenk Thanks for your contribution 🎉💙

**Pull request closed without merging**
